File: python/examples/repo_scan.py
"""Upload a repository archive (.zip) for a repo scan.

Run: DRALVIA_API_KEY=... python examples/repo_scan.py ./repo.zip
"""

import json
import os
import sys


from dralvia_sdk import DralviaClient


def main() -> int:
    if len(sys.argv) < 2:
        print("usage: python examples/repo_scan.py <path-to-repo.zip>")
        return 2
    archive_path = sys.argv[1]
    client = DralviaClient(api_key=os.environ.get("DRALVIA_API_KEY"))
    with open(archive_path, "rb") as fh:
        data = fh.read()
    result = client.scan_repo_archive(os.path.basename(archive_path), data, async_mode="1")
    print(f"scan id={result.get('id')} risk_level={result.get('risk_level')}")
    print(json.dumps(result, indent=2))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: js/examples/repo_scan.js
// Upload a repository archive (.zip) for a repo scan.
//
// Run: DRALVIA_API_KEY=... node examples/repo_scan.js ./repo.zip
import { readFile } from "node:fs/promises";
import { basename } from "node:path";
import { DralviaClient } from "../src/client.js";

const archivePath = process.argv[2];
if (!archivePath) {
  console.error("usage: node examples/repo_scan.js <path-to-repo.zip>");
  process.exit(2);
}

const client = new DralviaClient({ apiKey: process.env.DRALVIA_API_KEY });
const bytes = await readFile(archivePath);

const result = await client.scanRepoArchive(basename(archivePath), bytes, { async: "1" });
console.log(`scan id=${result.id} risk_level=${result.risk_level}`);
console.log(JSON.stringify(result, null, 2));
